slack.js

```diff
@@ -0,0 +1,363 @@
+        var allLeagues = league.getAllLeagues(config);
+        var leagueTargets = [];
+        var targetToLeague = {};
+        function add_target(l, target) {
+            leagueTargets.push(target);
+            targetToLeague[target] = l;
+        }
+        _.each(allLeagues, function(l) {
+            add_target(l, l.options.name);
+            _.each(l.options.also_known_as, function(aka) {
+                add_target(l, aka);
+            });
+        });
+
+        // Now fuzzy match them based on each arg in the message
+        var matches = [];
+        var args = message.text.split(" ");
+        _.each(args, function(arg) {
+            var results = fuzzy.rank_choices(arg.toLowerCase(), leagueTargets, true);
+            _.each(results, function(match) {
+                matches.push(match);
+            });
+        });
+        var bestMatches = fuzzy.findBestMatches(matches, true);
+        var possibleLeagues = {};
+        _.each(bestMatches, function(match) {
+            var l = targetToLeague[match[1]];
+            possibleLeagues[l.options.name] = l;
+        });
+        var matchingLeagueNames = _.keys(possibleLeagues);
+        if (matchingLeagueNames.length > 1) {
+            throw new Error("Ambiguous leagues.");
+        }
+        if (matchingLeagueNames.length == 1) {
+            l = _.values(possibleLeagues)[0];
+            message.league = l;
+            return l;
+        }
+
+        // If they didn't ask for a specific league, then we will use the channel
+        // to determine it
+        var channel = channels.byId[message.channel];
+        if (channel && channel.name) {
+            var target_league = config.channel_map[channel.name];
+            l = league.getLeague(target_league, config);
+            if (l) {
+                message.league = l;
+                return l;
+            }
+        }
+
+        return;
+    });
+}
+function requiresLeague(bot, message, config) {
+    return withLeague(bot, message, config).then(function(l) {
+        if (!l || !message.league) {
+            bot.reply(message, "This command requires you to specify the league you are interested in. Please include that next time");
+            throw new StopControllerError("No league specified");
+        }
+    });
+}
+
+
+//------------------------------------------------------------------------------
+// A controller wrapper.
+//------------------------------------------------------------------------------
+DEFAULT_HEARS_OPTIONS = {
+    middleware: []
+};
+function hears(options, callback) {
+    var self = this;
+    var options = _.extend({}, DEFAULT_HEARS_OPTIONS, options);
+    self.controller.hears(options.patterns, options.messageTypes, function(bot, message) {
+        return botExceptionHandler(bot, message, Q.fcall(function() {
+            message.player = users.getByNameOrID(message.user);
+            // This will occur if a new player uses a chesster command
```

Comment:
this could likely happen if the welcome becomes interactive. just something to keep in mind.